lightning/src/ln/channelmanager.rs

```diff
@@ -4140,7 +4173,7 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 					let mut htlc_msat_height_data = byte_utils::be64_to_array(htlc.value).to_vec();
 					htlc_msat_height_data.extend_from_slice(&byte_utils::be32_to_array(
 							self.best_block.read().unwrap().height()));
-					self.fail_htlc_backwards_internal(channel_state.take().unwrap(),
+					self.fail_htlc_backwards_internal(
```

Comment:
Thanks! Fixed this with 383ef941a1f99566bab8ffb5e81737b983663a78. Not sure if I overdid it, given that we'll have the unique peer´s  lock with #1507, which kind of removes the purpose of my refactor, instead of just taking the `channel_state` lock several times. I pushed this in a separate commit just in case you think I overdid this, so I can revert it easly.